src/test/java/com/teragrep/lsh_01/MultithreadingEndToEndTest.java

```diff
@@ -0,0 +1,139 @@
+        System.setProperty("properties.file", "src/test/resources/properties/multithreadingTest.properties");
+
+        // Start listening to HTTP-requests
+        program = new Thread(() -> Main.main(new String[] {}));
+        program.start();
+
+        Thread.sleep(3000); // wait for netty to start up
+
+        this.relpServer = new RelpServer();
+        this.relpServer.setUpDefault();
+    }
+
+    @AfterEach
+    void reset() {
+        this.relpServer.clear();
+    }
+
+    @AfterAll
+    void tearDown() {
+        System.clearProperty("properties.file");
+        this.relpServer.tearDown();
+        program.interrupt();
+    }
+
+    @Test
+    public void testMultithreadingTenClients() {
+        List<String> requestBodies = Collections.synchronizedList(new ArrayList<>());
+
+        HttpClient httpClient = HttpClient.newHttpClient();
+        NettyConfig nettyConfig = new NettyConfig();
+
+        final int clients = 10;
+        final int messagesPerClient = 100;
+
+        ExecutorService executor = Executors.newFixedThreadPool(clients);
+        List<Future<?>> futures = new ArrayList<>();
+
+        // Send messages in parallel
+        for (int i = 0; i < clients; i++) {
+            Future<?> future = executor.submit(() -> {
+                for (int j = 0; j < messagesPerClient; j++) {
+                    String requestBody = randomString();
+                    requestBodies.add(requestBody);
+
+                    HttpRequest request = HttpRequest
+                            .newBuilder(URI.create("http://" + nettyConfig.listenAddress + ":" + nettyConfig.listenPort)).POST(HttpRequest.BodyPublishers.ofString(requestBody)).build();
+
+                    CompletableFuture<HttpResponse<String>> response = httpClient
+                            .sendAsync(request, HttpResponse.BodyHandlers.ofString());
+
+                    // Assert that there is a successful response
+                    int statusCode = Assertions.assertDoesNotThrow(() -> response.get().statusCode());
+                    Assertions.assertEquals(200, statusCode);
+                }
+            });
+            futures.add(future);
+        }
+
+        // wait until all threads are done
+        for (Future<?> future : futures)
+            Assertions.assertDoesNotThrow(() -> future.get());
+
+        List<String> payloads = this.relpServer.payloads(); // get the results
+        Assertions.assertEquals(clients * messagesPerClient, payloads.size());
+
+        int loops = 0;
+        RFC5424Frame frame = new RFC5424Frame();
+        for (String payload : payloads) {
+            frame.load(new ByteArrayInputStream(payload.getBytes(StandardCharsets.UTF_8)));
+            Assertions.assertDoesNotThrow(frame::next);
+            Assertions.assertTrue(requestBodies.contains(frame.msg.toString())); // order of payloads can differ
+            loops++;
+        }
+        Assertions.assertEquals(payloads.size(), loops);
```

Comment:
please do not use payloads.size() because if none are received then this would pass silently

Reply:
Changed so that it now uses the expected amount of messages instead of the amount of received messages. Did this for Json and Regex splitting tests as well.